tests/integration/lifecycle/test_lifecycle_partitions.py

```diff
@@ -0,0 +1,390 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021-2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import textwrap
+from pathlib import Path
+
+import pytest
+import yaml
+
+import craft_parts
+from craft_parts import Action, ActionProperties, ActionType, Step
+
+basic_parts_yaml = textwrap.dedent(
+    """\
+    parts:
+      bar:
+        after: [foo]
+        plugin: nil
+
+      foo:
+        plugin: nil
+        source: a.tar.gz
+
+      foobar:
+        plugin: nil"""
+)
+
+
+@pytest.fixture(autouse=True)
+def setup_feature(enable_partitions_feature):
+    yield
+
+
+def test_basic_lifecycle_actions(new_dir, mocker):
```

Comment:
If this is the same as the regular lifecycle test with only the partitions parameters changed, maybe we could devise a way to use a common implementation in a future test refactoring round.

Reply:
+1.  As discussed offline, it isn't worth a grandiose refactoring for two features.  When we add a third feature, we can revisit how to best refactor the tests.